python/mobitex_fec_block.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2025 Fabian P. Schmidt <kerel@mailbox.org>
+#
+# This file is part of gr-satellites
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from itertools import batched, chain
+
+from gnuradio import gr
+import pmt
+
+from .mobitex_fec import decode, Status, unpack_2b
+
+
+class mobitex_fec(gr.basic_block):
+    """
+    Block to perform error-correction on Mobitex data blocks.
+
+    # Input
+    PDU with encoded Mobitex data block (20 * 12 bits)
+
+    # Output
+    PDU with decoded Mobitex data block (20 * 8 bits)
+
+    Output Tags:
+    - corrected_errors (int): Number of corrected bit errors
+    - uncorrected_errors (int): Number of uncorrectable bit errors
+    """
+
+    def __init__(self):
+        gr.basic_block.__init__(
+            self,
+            name='mobitex_fec',
+            in_sig=[],
+            out_sig=[])
+
+        self.message_port_register_in(pmt.intern('in'))
+        self.set_msg_handler(pmt.intern('in'), self.handle_msg)
+        self.message_port_register_out(pmt.intern('out'))
+
+    def handle_msg(self, msg_pmt):
+        meta = pmt.car(msg_pmt)
+        msg = pmt.cdr(msg_pmt)
+        if not pmt.is_u8vector(msg):
+            print('[ERROR] Received invalid message type. Expected u8vector')
+            return
+
+        block = pmt.u8vector_elements(msg)
+
+        if len(block) != 30:
+            print(
+                '[ERROR] Received invalid Mobitex data block, '
+                f'expected 30 bytes (got {len(block)} bytes).'
+            )
+            return
+
+        codewords = chain(*[
+            unpack_2b(bytes(item)) for item in batched(block, n=3)
+        ])
```

Comment:
It seems weird to use `chain` for this. I would do:
```python
codewords = [b for item in batched(block, n=3) for b in unpack_2b(bytes(item))]
```
(and then find a replacement for `batches`).